platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/SignedStateFileManagerWiring.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.wiring;
+
+import com.swirlds.common.system.status.PlatformStatusManager;
+import com.swirlds.common.system.status.actions.StateWrittenToDiskAction;
+import com.swirlds.common.wiring.TaskScheduler;
+import com.swirlds.common.wiring.wires.input.BindableInputWire;
+import com.swirlds.common.wiring.wires.input.InputWire;
+import com.swirlds.common.wiring.wires.output.OutputWire;
+import com.swirlds.platform.components.appcomm.AppCommunicationComponent;
+import com.swirlds.platform.event.preconsensus.PreconsensusEventWriter;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import com.swirlds.platform.state.signed.SignedStateFileManager;
+import com.swirlds.platform.state.signed.StateDumpRequest;
+import com.swirlds.platform.state.signed.StateSavingResult;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * The wiring for the {@link SignedStateFileManager}
+ *
+ * @param outputWire      the output wire
+ * @param saveStateToDisk the input wire for saving the state to disk
+ * @param dumpStateToDisk the input wire for dumping the state to disk
+ */
+public record SignedStateFileManagerWiring(
+        @NonNull OutputWire<StateSavingResult> outputWire,
+        @NonNull InputWire<ReservedSignedState> saveStateToDisk,
+        @NonNull InputWire<StateDumpRequest> dumpStateToDisk) {
```

Comment:
Nit. We've not really established a convention on this, but perhaps we should put input wires first then output wires.

Reply:
In the interest of merging this now without requiring the reviews of 3 teams again, I will do it in a subsequent PR